src/engine/TransitivePathBase.cpp

```diff
@@ -354,13 +354,15 @@ std::shared_ptr<TransitivePathBase> TransitivePathBase::bindLeftOrRightSide(
   // subtrees. This has the effect that the `TransitivePathBinSearch` will
   // never re-sort an index scan (which should not happen because we can just
   // take the appropriate index scan in the first place).
+  bool useBinSearch = dynamic_cast<const TransitivePathBinSearch*>(this);
   std::vector<std::shared_ptr<TransitivePathBase>> candidates;
-  candidates.push_back(TransitivePathBase::makeTransitivePath(
-      getExecutionContext(), subtree_, lhs, rhs, minDist_, maxDist_));
+  candidates.push_back(makeTransitivePath(getExecutionContext(), subtree_, lhs,
+                                          rhs, minDist_, maxDist_,
+                                          useBinSearch));
   for (const auto& alternativeSubtree : alternativeSubtrees()) {
-    candidates.push_back(TransitivePathBase::makeTransitivePath(
-        getExecutionContext(), alternativeSubtree, lhs, rhs, minDist_,
-        maxDist_));
+    candidates.push_back(makeTransitivePath(getExecutionContext(),
+                                            alternativeSubtree, lhs, rhs,
+                                            minDist_, maxDist_, useBinSearch));
   }
```

Comment:
This is so unit tests with bound variables properly check both implementations and not just the one used by the configuration

Reply:
Ah, so in unit tests, if I say "I want binSearch", then place would overwrite the setting again?

Reply:
Yes, exactly